core/network/src/on_demand.rs

```diff
@@ -152,8 +151,8 @@ impl<B: BlockT, E> OnDemand<B, E> where
 	}
 
 	/// Sets weak reference to network service.
-	pub fn set_service_link(&self, service: Weak<E>) {
-		self.core.lock().service = service;
+	pub fn set_network_sender(&self, network_sender: NetworkChan) {
+		self.core.lock().network_sender = Some(network_sender);
```

Comment:
`self.core` holds a `NetworkChan` but methods like `maintain_peers` are also taking one by reference. Is one unnecessary?